client/landing/stepper/declarative-flow/internals/hooks/use-launchpad-decider/index.tsx

```diff
@@ -22,42 +24,38 @@ interface SiteProps {
 }
 
 export const useLaunchpadDecider = ( { exitFlow, navigate }: Props ) => {
-	// placeholder field for the experiment assignment
-	const showCustomerHome = false;
-	const sessionId = getSessionId();
-	let launchpadStateOnSkip: null | 'skipped' = null;
-	if ( showCustomerHome ) {
-		launchpadStateOnSkip = 'skipped';
+	const { getIntent } = useSelect( ( select ) => select( ONBOARD_STORE ) as OnboardSelect, [] );
+	const intent = getIntent();
+	const site = useSite();
+	// site intent is not set until we exit site setup flow so we set it here with onboarding store
+	// as it is used in shouldShowLaunchpadFirst. Note that we are not setting the site intent
+	// here to be saved server-side, we are only setting client-side so that we can use it in
+	// shouldShowLaunchpadFirst.
+	if ( site && site.options && site.options.site_intent === '' ) {
+		site.options.site_intent = intent;
```

Comment:
I think it's a bad idea to mutate the site object. For example, I almost added a function that cancels updating the site intent via the API when the site object already has the value. My goals was to save on requests.

I was playing around in [this PR](https://github.com/Automattic/wp-calypso/pull/101797/files#diff-614cccce787b17e46e84a16af7865700992dfd374e701239023fd2e1feaa12a1R227-R230). 

Especially, this object is coming from a store, which makes it almost a global that is used in many places.

We should pass a new object to the function instead 

```js
shouldShowLaunchpadFirst( {...site, options: {
	...site.options, 
	site_intent: intent
});
```

Reply:
Make sense. Thanks for the not, that's definitely a cleaner more robust solution.

Reply:
I would keep the code comment, it was quite helpful!